### PR TITLE
fix(proxy-cache): changes age param

### LINKED ISSUE #12787: Bug in proxy-cache plugin not returning age header
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Kong version (`$ kong version`)

Kong Enterprise 3.6.1.1

### Current Behavior

I have the proxy-cache plugin running (with the _response_headers_ parameters on default) and I do only get back _x-cache-status_ and _x-cache-key_ but not the _age_ header.

By default it should return _age_ and I don't specifically set those values and I also tried to force the _age_ to _true_ but that didn't make any difference.

I checked in both 3.5 (using Kong Gateway in docker) and 3.6.1.1 (using the Kong Gateway Operator in Kubernetes) and it was the same behaviour.

### Expected Behavior

I expect a header to be present called age with a number indicating the seconds the request is "old".

### Steps To Reproduce

1. add proxy-cache plugin to a route
2. run curl command with -v and inspect headers

### Anything else?

This is my Kubernetes resource:

```yaml
apiVersion: configuration.konghq.com/v1
kind: KongPlugin
metadata:
  name: my-cache-1min
  namespace: my-api-v4
config:
  cache_ttl: 60
  content_type:
    - text/plain
    - application/json
    - application/json; charset=utf-8
  request_method:
    - GET
    - HEAD
  response_code:
    - 200
  ignore_uri_case: true
  strategy: memory
plugin: proxy-cache
```

### PR DESCRIPTION
**Pull request by joelact**
### Summary

The age parameter on the schema was in lower case which caused the header Age to appear only when the kong debug option was enabled. This commit changes the schema parameter from age to Age.

### Checklist

- [x] The Pull Request has tests
- [X] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/README.md)
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

Fix #12787

### CODE REVIEW

**Comment by jschmid1** [comment]
Thanks for your contribution @joelact.
While a headers are case-insensitive according to the [RFC](https://www.rfc-editor.org/rfc/rfc9110.html#name-field-names) it's a contention to have them Title-cased.

Your change includes a schema change, which requires additional steps to ensure backwards compatibility with older Kong versions. If you choose to follow-up on those, I'm happy to assist you in doing so.
Alternatively, we'll pick up the PR and continue from here. Up to you @joelact

**Comment by joelact** [comment]
Hey @jschmid1. Thanks for taking a look at this.
I want to continue working on this PR so can you tell me the next steps to get this going?

**Comment by jschmid1** [comment]
Sure @joelact, Your change qualifies as a "rename" of a field. To ensure compatibility to older versions when running Kong in hybrid-mode  you have to teach the Kong how to manipulate the plugin config so that older versions still understand it.

You can take a look at the compatibility checkers here -> https://github.com/Kong/kong/blob/master/kong/clustering/compat/checkers.lua

and for testing check this file -> https://github.com/Kong/kong/blob/master/spec/02-integration/09-hybrid_mode/09-config-compat_spec.lua

**Comment by joelact** [comment]
Hey @jschmid1! Sorry for the delay in completing the mentioned actions. When possible can you review this again?

File: kong/plugins/proxy-cache/clustering/compat/response_headers_translation.lua [hunk 1]
```diff
@@ -0,0 +1,12 @@
+local function adapter(config_to_update)
+    if config_to_update.response_headers["Age"] ~= nil then
+        config_to_update.response_headers.age = config_to_update.response_headers["Age"]
```
Comment (jschmid1) [inline kong/plugins/proxy-cache/clustering/compat/response_headers_translation.lua:3]: This currently only sets the value of the `Age` field to the `age` field. We'd also have to remove the `Age` field from the adapted configuration.

```suggestion
        config_to_update.response_headers.age = config_to_update.response_headers["Age"]
        config_to_update.response_headers["Age"] = nil
```

File: spec/02-integration/09-hybrid_mode/09-config-compat_spec.lua [hunk 2]
```diff
@@ -426,7 +426,7 @@ describe("CP/DP config compat transformations #" .. strategy, function()
 
       describe("response-ratelimiting plugin", function()
         it("translates standardized redis config to older response-ratelimiting structure", function()
-          -- [[ 3.6.x ]] --
```
Comment (jschmid1) [inline spec/02-integration/09-hybrid_mode/09-config-compat_spec.lua:429]: This change seems unwanted

**Review by jschmid1 — DISMISSED** [review]
Sorry for the late reponse @joelact. There is one last thing we need to address. Other than that we're good to go. Thanks for all the effort, really appreciated 👍🏼

File: spec/02-integration/09-hybrid_mode/09-config-compat_spec.lua (hunk 2 above)
Comment (joelact) [reply spec/02-integration/09-hybrid_mode/09-config-compat_spec.lua]: reverted. thanks

File: kong/plugins/proxy-cache/clustering/compat/response_headers_translation.lua (hunk 1 above)
Comment (joelact) [reply kong/plugins/proxy-cache/clustering/compat/response_headers_translation.lua:3]: removed `Age` field. thanks

**Comment by jschmid1** [comment]
Looks good, waiting for the CI to turn green. 👍🏼

**Comment by jschmid1** [comment]
From https://github.com/Kong/kong/actions/runs/9221338444/job/25446686155?pr=12812

> File (kong/plugins/proxy-cache/clustering/compat/response_headers_translation.lua) not found in rockspec (kong-3.8.0-0.rockspec)


You need to add this file to the rockspec.

**Review by liverpool8056 — APPROVED** [review]

**Review by jschmid1 — APPROVED** [review]

**Comment by jschmid1** [comment]
Thanks for your contribution, much appreciated @joelact

**Pull request merged**